### PR TITLE
SASS-3235 update library dependencies

### PR DESCRIPTION
**Pull request by rtna-hmrc**
# Checklist

Jira subtask: [SASS-3370](https://jira.tools.tax.service.gov.uk/browse/SASS-3370)

## Developer: Before Asking Others To Review Your Code

Have you ...

- [x]  Had a QA verify that all acceptance criteria have been met, with comment(s) in the ticket? 
- [x]  Hosted a group review of your implementation?
- [x]  Rebased against `main`? 
- [x]  Ensured that your commit(s) contain the ticket key?
- [x]  Ensured that your branch is named after the ticket key?
- [x]  Successfully run the unit tests?
- [x]  Successfully run the integration tests?
- [x]  Successfully run the journey tests? If not, please add a comment explaining why it wasn't necessary.
- [x]  Ensured that the PR job has passed.

## Reviewer: Before Reviewing

Have you ...

- [x]  Pulled the branch down?
- [x]  Assigned yourself to this PR?
- [x]  Moved the Jira *subtask* to “in review”?
- [x]  Ensured that it is rebased against `main`?

## Developer: After Approval

Have you ...

- [x]  Rebased against `main`?
- [ ]  Moved the Jira *subtask* to “in pipeline”?

## Developer: After Merging

Have you ...

- [ ]  Deleted the branch?
- [ ]  Scheduled a demo? If not, please add a comment.

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -19,12 +19,12 @@ import sbt._
...
 
-  private val bootstrapPlay28Version = "6.4.0"
-  private val hmrcMongoPlayVersion = "0.68.0"
+  private val bootstrapPlay28Version = "7.1.0"
+  private val hmrcMongoPlayVersion = "0.71.0"
 
   val compile: Seq[ModuleID] = Seq(
     "uk.gov.hmrc"                   %% "bootstrap-frontend-play-28" % bootstrapPlay28Version,
-    "uk.gov.hmrc"                   %% "play-frontend-hmrc"         % "3.8.0-play-28",
+    "uk.gov.hmrc"                   %% "play-frontend-hmrc"         % "3.23.0-play-28",
```
Comment (salman-a-dev) [inline project/AppDependencies.scala:27]: Should this be 3.24.0-play-28 instead of 3.23.0-play-28?

**Review by salman-a-dev — COMMENTED** [review]
To upgrade to scala 2.13.8. 

- Silencer plugin needs to be updated to version 1.7.9
- sbt-coverage needs to be updated to version 1.9.3

This however causes tests to break and would need to be looked further (This will most likely need to be done at some point). Income tax pensions backend can be updated to use the versions specified above and all tests should pass. Perhaps Sean (@erasmos-hmrc) can provide some insight on how to proceed.

File: project/AppDependencies.scala (hunk 1 above)
Comment (rtna-hmrc) [reply project/AppDependencies.scala:27]: hmmm where did you see that version? I was based on the [following page](https://catalogue.tax.service.gov.uk/service/income-tax-pensions-frontend):

image

Comment (rtna-hmrc) [reply project/AppDependencies.scala:27]: I see... do you know the difference between those 2? and why the discrepancy?

Comment (salman-a-dev) [reply project/AppDependencies.scala:27]: Between the picture and the link to whats running where?

Comment (erasmos-hmrc) [reply project/AppDependencies.scala:27]: As agreed in the group review, @rtna-hmrc , we'll postpone the upgrade of Scala; it's OK for this ticket to keep it at 2.12.

**Review by salman-a-dev — APPROVED** [review]

**Comment by rtna-hmrc** [comment]
No demo needed, as no new functionality has been added

**Comment by erasmos-hmrc** [comment]
Agreed.

**Pull request merged**